packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolGetEntity.ts

```diff
@@ -0,0 +1,97 @@
+import { useLazyQuery } from "@apollo/client";
+import { useCallback } from "react";
+import { getEntityTypeRootedSubgraphQuery } from "../../../../graphql/queries/ontology/entity-type.queries";
+
+import {
+  GetEntityTypeRootedSubgraphQuery,
+  GetEntityTypeRootedSubgraphQueryVariables,
+  Query,
+  QueryKnowledgeEntityArgs,
+} from "../../../../graphql/apiTypes.gen";
+import { getKnowledgeEntity } from "../../../../graphql/queries/knowledge/entity.queries";
+import { GetEntityMessageCallback } from "./knowledge-shim";
+
+export const useBlockProtocolGetEntity = (): {
+  getEntity: GetEntityMessageCallback;
+} => {
+  const [getEntityTypeRootedSubgraphFn] = useLazyQuery<
+    GetEntityTypeRootedSubgraphQuery,
+    GetEntityTypeRootedSubgraphQueryVariables
+  >(getEntityTypeRootedSubgraphQuery, {
+    /** @todo reconsider caching. This is done for testing/demo purposes. */
+    fetchPolicy: "no-cache",
+  });
+
+  const [getEntityFn] = useLazyQuery<Query, QueryKnowledgeEntityArgs>(
+    getKnowledgeEntity,
+    {
+      /** @todo reconsider caching. This is done for testing/demo purposes. */
+      fetchPolicy: "no-cache",
+    },
+  );
+
+  const getEntity = useCallback<GetEntityMessageCallback>(
+    async ({ data }) => {
+      if (!data) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "'data' must be provided for getEntity",
+            },
+          ],
+        };
+      }
+
+      const { entityId } = data;
+
+      const { data: entityResponseData } = await getEntityFn({
+        query: getKnowledgeEntity,
+        variables: { entityId },
+      });
+
+      if (!entityResponseData) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "Error calling getEntity",
+            },
+          ],
+        };
+      }
+
+      const { knowledgeEntity } = entityResponseData;
+
+      const { data: entityTypeResponseData } =
+        await getEntityTypeRootedSubgraphFn({
```

Comment:
Wonder if it's worth a TODO to say that these two should be one query to get an EntityRootedSubgraph

Reply:
Good idea, is there a task for the upcoming GQL refactor work?

Reply:
Probably going to end up encapsulate in [this task](https://app.asana.com/0/1200211978612931/1203089535761796/f) (might rework its title and description)

Reply:
Added the todo in https://github.com/hashintel/hash/pull/1160/commits/03b165ace4db95f35a3c55c7b788470c9413087b

Reply:
Added the todo in https://github.com/hashintel/hash/pull/1160/commits/03b165ace4db95f35a3c55c7b788470c9413087b